test/test_accelerator.py

```diff
@@ -0,0 +1,70 @@
+# Owner(s): ["module: tests"]
+
+import unittest
+
+import torch
+from torch.testing._internal.common_utils import (
+    run_tests,
+    TEST_CUDA,
+    TEST_XPU,
+    TestCase,
+)
+
+
+TEST_ACCELERATOR = TEST_CUDA or TEST_XPU
+
+
+class TestAccelerator(TestCase):
+    def test_generic_device_behavior(self):
+        if not TEST_ACCELERATOR:
+            self.assertIsNone(torch.current_accelerator())
+            return
+        current_device = torch.current_device()
+        torch.set_device(current_device)
+        self.assertEqual(current_device, torch.current_device())
+        if torch.cuda.is_available():
+            self.assertTrue(torch.has_accelerator())
+            self.assertEqual(torch.current_accelerator(), "cuda")
+        if torch.xpu.is_available():
+            self.assertTrue(torch.has_accelerator())
+            self.assertEqual(torch.current_accelerator(), "xpu")
```

Comment:
Please add device to the input parameter and simplify the logic a little bit.

Reply:
done.